sql/builtin_functions.cc

```diff
@@ -3439,6 +3440,54 @@ bind(const std::vector<BoundSqlExpression> & args,
     }
 }
 
+
+BoundFunction get_mime_type(const std::vector<BoundSqlExpression> & args)
+{
+    checkArgsSize(args.size(), 1);
+    auto outputInfo = std::make_shared<Utf8StringValueInfo>();
+
+    return {[=] (const std::vector<ExpressionValue> & args,
+                 const SqlRowScope & scope) -> ExpressionValue
+            {
+                const char *mime;
+                magic_t magic;
+
+                magic = magic_open(MAGIC_NONE);
+                if(magic == NULL) {
+                    throw ML::Exception("Unable to initialize the magic library");
+                }
+
+                if(magic_load(magic, NULL) != 0) {
+                    magic_close(magic);
+                    throw ML::Exception("Error loading magic database - %s\n", magic_error(magic));
+                }
+
+                if(!args[0].isAtom())
+                    throw MLDB::Exception("Mime type extraction requires that an atomic value "
+                            "of type BLOB is passed to it. Use the fetcher function to get the file blob.");
+
+                const CellValue & input = args[0].getAtom();
+
+                if(!input.isBlob())
+                    throw MLDB::Exception("Mime type extraction requires that an atomic value "
+                            "of type BLOB is passed to it. Use the fetcher function to get the file blob.");
+
+                const unsigned char * data = input.blobData();
+                const size_t len = input.blobLength();
+
+                mime = magic_buffer(magic, data, len);
+
+                Utf8String str(mime);
+                magic_close(magic);
```

Comment:
What happens if this doesn't get called? Does it leak? If so we need to wrap it into an RAII structure.